### PR TITLE
Table border radius

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

Fixed border-radius for table => table should always have 4px border-radius all around
- If footer then footer deals with bottom border-radius if not table handles it
- Added hasRadius to main table container to prevent weird borders overflowing
- Overflow hidden on TableBox to prevent shadows to overflow border-radius

## Screenshots 

Capture d’écran 2021-10-05 à 16 14 09

Capture d’écran 2021-10-05 à 16 14 16

Capture d’écran 2021-10-05 à 16 12 01

### CODE REVIEW

**Review by maevalienard — APPROVED** [review]
Well done Julie! Thank you :D

File: packages/strapi-parts/src/Table/Table.js [hunk 1]
```diff
@@ -46,6 +48,7 @@ const ScrollContainer = styled(Box)`
 export const Table = ({ colCount, rowCount, footer, ...props }) => {
   const tableRef = useRef(null);
   const [overflowing, setOverflowing] = useState();
+  console.log(footer);
```
Comment (HichamELBSI) [inline packages/strapi-parts/src/Table/Table.js:51]: Can you remove the console.log please

File: packages/strapi-parts/src/Table/Table.js [hunk 2]
```diff
@@ -70,8 +73,8 @@ export const Table = ({ colCount, rowCount, footer, ...props }) => {
   }, []);
 
   return (
-    <Box shadow="tableShadow">
-      <TableBox background="neutral0" overflowing={overflowing}>
+    <Box shadow="tableShadow" hasRadius>
+      <TableBox footer={footer} background="neutral0" overflowing={overflowing}>
```
Comment (HichamELBSI) [inline packages/strapi-parts/src/Table/Table.js:77]: Can you rename the TableBox props please:

Something like 
```js
fullBorderRadius={footer}
or
withBorderRadius={footer}
or 
hasFullBorderRadius={footer}
🤷🏽‍♂️
```

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-parts/src/Table/Table.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/strapi-parts/src/Table/Table.js:51]: oh sure will do sorry!

Comment (soupette) [reply packages/strapi-parts/src/Table/Table.js]: We should add an eslint rule for this, WDYT?

File: packages/strapi-parts/src/Table/Table.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-parts/src/Table/Table.js:76]: I'd prefer something that starts by `with` ....

Comment (ronronscelestes) [reply packages/strapi-parts/src/Table/Table.js:76]: let's go for with

**Review by mfrachet — CHANGES_REQUESTED** [review]
I'll take a look tomorrow, I'm really not fan of adding this kind of props

**Pull request closed without merging**